crates/circuit/src/operations.rs

```diff
@@ -688,9 +663,12 @@ impl StandardGate {
     }
 
     // These pymethods are for testing:
-    pub fn _to_matrix(&self, py: Python, params: Vec<Param>) -> Option<PyObject> {
-        self.matrix(&params)
-            .map(|x| x.into_pyarray_bound(py).into())
+    pub fn _to_matrix<'py>(
+        &'py self,
+        py: Python<'py>,
+        params: Vec<Param>,
+    ) -> Option<Bound<'py, PyArray2<Complex64>>> {
+        self.matrix(&params).map(|x| x.into_pyarray(py))
     }
```

Comment:
I don't think the lifetime bound on `self` needs to live as long as `'py`?  Not that I think it matters, since this method is marked "just for testing".

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/13577/commits/a2813401119cd80360cfb6891ea30c41525d60f8